### ISSUE TITLE
Misshaped character 'B' with Roboto Font

### ISSUE THREAD

**Issue by timrosskamp** [gregdavill/KiBuzzard#88]
The character 'B' with the Font "Roboto" seems to be problematic. This could also be a problem with the font itself, but the Roboto Font is very popular, so it might be worth looking into it.

Edit: I just noticed something. I've been using a previous version of this plugin on older projects and the problem with the B character wasn't present. This could be something in the `1.2.0` change. I'm not certain, but I think I've been using `1.1.0` before.

Edit 2: I just tested this with version `1.0.0` and I don't get this problem. The character 'B' looks fine. So this looks to me like a problem introduced in `1.2.0`.

Roboto Font: https://fonts.google.com/specimen/Roboto

![Bildschirmfoto 2022-08-16 um 17 22 30](https://user-images.githubusercontent.com/23245893/184917715-407cbab5-896f-43fb-84a6-8fdf819e3608.png)
![Bildschirmfoto 2022-08-16 um 17 25 52](https://user-images.githubusercontent.com/23245893/184918138-bbe23285-c527-4eb2-b67b-f5b793c54443.png)

**Comment by yaqwsx** [comment]
I am experiencing similar problems with UbuntuMono-B font (the T is cut):

![Screenshot from 2022-08-16 19-18-44](https://user-images.githubusercontent.com/1590880/184940088-a5fb3c7c-c84d-4a91-9c45-2013dd9c633d.png)

**Comment by gregdavill** [comment]
This looks to be a regression.
There were some performance improvements in svg2mod, with respect to how the polygons are sliced. These changes might be the cause of this type of rendering error.

**Comment by electrified** [comment]
Hi, Using the font Exo Black available here https://fonts.google.com/specimen/Exo

I see the same behaviour in version 1.0.0, 1.1.0 and 1.2.0.

Expected
![image](https://user-images.githubusercontent.com/103082/192890240-7afa7387-575a-4e35-925b-c71866e87c8a.png)

Actual
![image](https://user-images.githubusercontent.com/103082/192890458-faba6de8-a7e3-479a-8734-a0fabc634442.png)

**Comment by Sodium-Hydrogen** [comment]
This is related to svg2mod/svg2mod#51.
I am aware and have a fix (svg2mod/svg2mod#53).
I am just finalizing polygon support for svgs then that fix will be merged into svg2mod.

**Comment by Sodium-Hydrogen** [comment]
Version 1.2 of svg2mod is out with the fix for this.

I did rework the way styles are handled to support svgs without explicitly defined styles which does seem to break kibuzzard but this change in [buzzard.py 156](https://github.com/gregdavill/KiBuzzard/blob/00168be6342af39940274283d46ba4e55a08c224/KiBuzzard/buzzard/buzzard.py#L156) seems to fix the crashing I was experiencing:

```diff
- offset_text.style += "stroke-width:0.2;"
+ offset_text.style["stroke-width"] = 0.2
```

**Comment by gregdavill** [comment]
@Sodium-Hydrogen I've updated the svg2mod dependency (https://github.com/gregdavill/KiBuzzard/pull/94). However when I try the example from above I get a different error, it seems the polygon ends up as self-intersecting. Or atleast ends up with overlapping parallel lines which seem to cause some issues.
![Screenshot from 2022-11-20 22-55-12](https://user-images.githubusercontent.com/344310/202901917-a67157a7-dd45-4327-a5fe-9b23786f1d75.png)
![Screenshot from 2022-11-20 22-55-03](https://user-images.githubusercontent.com/344310/202901962-1e3c9326-94df-4603-a074-0af6428d30ff.png)

In the PCB editor the second cutout is missing. However in the 3d view it's all okay.

![Screenshot from 2022-11-20 22-53-06](https://user-images.githubusercontent.com/344310/202901901-eeb64d60-8609-4fdf-9166-427a16743856.png)

I'll poke into this a bit more myself, but thought I'd mention it in case you've seen it before and there is an easy fix.

**Comment by gregdavill** [comment]
I'll think about how to fix this in a more elegant way. But a quick fix for the `B` is to flip the order the segments are inlined.

```diff
diff --git a/KiBuzzard/deps/svg2mod/svg2mod/svg2mod.py b/KiBuzzard/deps/svg2mod/svg2mod/svg2mod.py
index f6e63dfd9..92c96cce0 100755
--- a/KiBuzzard/deps/svg2mod/svg2mod/svg2mod.py
+++ b/KiBuzzard/deps/svg2mod/svg2mod/svg2mod.py
@@ -303,7 +303,7 @@ class PolygonSegment:
 
         logger.debug( "  Inlining {} segments...".format( len( segments ) ) )
 
-        segments.sort(reverse=True, key=lambda h: h.bbox[1].y)
+        segments.sort(reverse=False, key=lambda h: h.bbox[1].y)
 
         all_segments = segments[ : ] + [ self ]
         insertions = []
```

This achieves a much cleaner inlining, without the knot that's currently created. (manually separated here to highlight the insertion points)
![Screenshot from 2022-11-22 21-32-35](https://user-images.githubusercontent.com/344310/203300053-b86cb82f-b1ef-454b-8550-c821429bf3b5.png)

![Screenshot from 2022-11-22 21-34-56](https://user-images.githubusercontent.com/344310/203299997-e36aa56b-7f69-43c6-8015-7be98462f5a1.png)

I'm not seeing any downsides to the sorting reversal, unless it tried to perform an insertion to the top, rather than the bottom, then we're back where we started.

**Comment by Sodium-Hydrogen** [comment]
Nice patch!

I think your fix works for now, but I'm working on svg even-odd support so I will have to get this working for 100% of cases.

Looking at the English alphabet I think your fix will achieve enough functionality that it will be unlikely anyone else will run into this problem before a comprehensive solution exists.

**Comment by gregdavill** [comment]
@timrosskamp Can you confirm that this fix works? It's currently included in a 1.3.0-rc release you can install https://github.com/gregdavill/KiBuzzard/releases/tag/1.3.0-rc
